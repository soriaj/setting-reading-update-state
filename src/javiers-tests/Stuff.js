import React from 'react'

class Stuff extends React.Component{
   constructor(props){
      super(props)
      this.state = {
         firstName: 'Javier',
         lastName: 'Soria',
         test: null
      }
   }
   handleClick = () => {
      this.timeout = setTimeout(() => {
         this.setState({
            firstName: 'JAVIER',
            lastName: 'SORIA',
            test: 'test'
         })
      }, 2000)
   }
   fullName(){
      return `Hello my name is ${this.state.firstName} ${this.state.lastName} ${this.state.test}`
   }
   componentWillUnmount(){
      clearTimeout(this.timeout)
   }
   render(){
      return(
         <div>
            <p>{this.fullName()}</p>
            <button 
               onClick={this.handleClick}>
               Click Here!
            </button>
         </div>
      )
   }
}

export default Stuff;