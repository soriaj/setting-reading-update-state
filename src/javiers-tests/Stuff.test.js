import React from 'react';
import ReactDOM from 'react-dom';
import Stuff from './Stuff';
import Testrenderer from 'react-test-renderer';

describe('Testing Stuff Component', () => {
   it('Smoke Test: Renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<Stuff />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('Test Render: renders the UI as expected', () => {
      const tree = Testrenderer
        .create(<Stuff />)
        .toJSON();
      expect(tree).toMatchSnapshot();  
    });
    
    // Additional Tests Can Go Here
})
