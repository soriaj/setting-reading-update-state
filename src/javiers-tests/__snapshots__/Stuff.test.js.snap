// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Stuff Component Test Render: renders the UI as expected 1`] = `
<div>
  <p>
    Hello my name is Javier Soria null
  </p>
  <button
    onClick={[Function]}
  >
    Click Here!
  </button>
</div>
`;
